### ISSUE TITLE
NullPointerException when overriding testKit from ScalaTestWithActorTestKit

### ISSUE THREAD

**Issue by ccerbusca** [apache/incubator-pekko#1140]
While migrating a project to Pekko, I noticed that some of the tests started failing with `NullPointerException`s.

On further investigation, it was because I was overriding the TestKit inside those tests.
Here's a simple reproducer:

```scala
import org.apache.pekko.actor.testkit.typed.scaladsl.{ActorTestKit, ScalaTestWithActorTestKit}
import org.scalatest.funsuite.AnyFunSuiteLike
 
class Test extends ScalaTestWithActorTestKit with AnyFunSuiteLike {
 
  override val testKit = ActorTestKit()
 
  test("test") {
    assert(true)
  }
 
}
```

I think the issue seems to be this line, where the `val` is not lazy, leading to a NullPointerException.

https://github.com/apache/incubator-pekko/blob/28314e679885519f6458ebf10f9a181d5299c83a/actor-testkit-typed/src/main/scala/org/apache/pekko/actor/testkit/typed/scaladsl/ScalaTestWithActorTestKit.scala#L94

I understand that in this case I should have passed the `TestKit` to the `ScalaTestWithActorTestKit` constructor (which I do now), but having the ability to override `testKit`, makes it a possible usage for other people too.

**Comment by mdedetrich** [comment]
I don't think this is generally solvable due to how trait initialization works in Scala 2, long story short trait composition in Scala 2 is not commutative which means that the order of when you mix in traits can effect initialization which can lead to `NullPointerException` in cases like this.

The solution to this problem is to use `lazy val`/`def`. Also afaik this is not an issue in Scala 3 where the compiler will re-order the fields in traits to make sure its initialized correctly.

**Comment by jxnu-liguobin** [comment]
Can we add documentation to explain it？

**Comment by mdedetrich** [comment]
> Can we add documentation to explain it？Although this is an initialization pit.

Sure, although I wouldn't spend too much time on it because as said previously it really has nothing specifically to do with Pekko.

I would propose adding a single line in the docs briefly explaining the problem containing a link to an official Scala 2 source explaining this.

I will also remove the bug label for this reason.
